modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -719,6 +738,65 @@ private void _addObjectDefinitionResources(
 		}
 	}
 
+	private Set<String> _getAccountEntryRestrictiveObjectRelationshipsNames(
+		ObjectRelationship[] objectRelationships,
+		com.liferay.object.model.ObjectDefinition
+			serviceBuilderObjectDefinition) {
+
+		if ((objectRelationships == null) ||
+			(serviceBuilderObjectDefinition == null) ||
+			!StringUtil.equals(
+				serviceBuilderObjectDefinition.getName(), "AccountEntry")) {
+
+			return Collections.emptySet();
+		}
+
+		Set<String> accountEntryRestrictiveObjectRelationshipsNames =
+			new HashSet<>();
+
+		for (ObjectRelationship objectRelationship : objectRelationships) {
+			if (!StringUtil.equals(
+					String.valueOf(objectRelationship.getType()),
+					ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
+
+				continue;
+			}
+
+			com.liferay.object.model.ObjectDefinition
+				serviceBuilderObjectDefinition2 =
+					_objectDefinitionLocalService.fetchObjectDefinition(
+						objectRelationship.getObjectDefinitionId2());
+
+			if ((serviceBuilderObjectDefinition2 == null) ||
+				!serviceBuilderObjectDefinition2.getAccountEntryRestricted()) {
+
+				continue;
+			}
+
+			com.liferay.object.model.ObjectRelationship
+				serviceBuilderObjectRelationship =
+					_objectRelationshipLocalService.
+						fetchObjectRelationshipByObjectDefinitionId(
+							serviceBuilderObjectDefinition.
+								getObjectDefinitionId(),
+							objectRelationship.getName());
+
+			if (serviceBuilderObjectRelationship == null) {
+				continue;
+			}
+
+			if (serviceBuilderObjectRelationship.getObjectFieldId2() ==
```

Comment:
Hey @devpedrotavares 

Whats happen if the user updating Account Restriction to another objectFieldId?